src/cmap/connect.ts

```diff
@@ -502,7 +502,7 @@ function makeSocks5Connection(options: MakeConnectionOptions, callback: Callback
 function connectionFailureError(type: ErrorHandlerEventName, err: Error) {
   switch (type) {
     case 'error':
-      return new MongoNetworkError(err);
+      return new MongoNetworkError('error', { cause: err });
```

Comment:
I was going to suggest stealing the message from the error here but I think this is our aggregate error case so the message will be an empty string.

Maybe we allow the string argument to be nullish, then buildMessage's job will be to obtain it from the cause, or we make this say "error during connection establishment"

Reply:
I'm a fan of the "make all consumers specify the error message".  It's more explicit and it makes it easier to reason about what error messages are for given errors.

If we want to handle aggregate errors, there's nothing stopping us from using `buildErrorMessage` wherever we want:

```typescript
      return new MongoNetworkError(MongoError.buildErrorMessage(err), { cause: err });
```